Updates to `ldc2.conf` are in line with discussion in https://github.com/ldc-developers/ldc2.snap/issues/102.

I'd like to also update to LLVM 10.0.1 but I keep getting build errors for `SPIRV-LLVM-Translator`:

```
[2336/3819] Building CXX object projects/SPIRV-LLVM-Translator/lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/VectorComputeUtil.cpp.o
FAILED: /usr/bin/c++   -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Iprojects/SPIRV-LLVM-Translator/lib/SPIRV -I/root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV -Iinclude -I/root/parts/llvm/src/llvm/include -I/root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/include -I/root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/libSPIRV -I/root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/Mangler -static-libstdc++ -fPIC -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wno-maybe-uninitialized -Wdelete-non-virtual-dtor -Wno-comment -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG    -fno-exceptions -fno-rtti -std=c++14 -MMD -MT projects/SPIRV-LLVM-Translator/lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/VectorComputeUtil.cpp.o -MF projects/SPIRV-LLVM-Translator/lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/VectorComputeUtil.cpp.o.d -o projects/SPIRV-LLVM-Translator/lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/VectorComputeUtil.cpp.o -c /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/VectorComputeUtil.cpp
/root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/VectorComputeUtil.cpp:68:69: error: specialization of 'template<class Ty1, class Ty2, class Identifier> void SPIRV::SPIRVMap<Ty1, Ty2, Identifier>::init()' in different namespace [-fpermissive]
 template <> inline void SPIRVMap<VCRoundMode, VCFloatControl>::init() {
                                                                     ^
In file included from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/libSPIRV/SPIRVOpCode.h:43:0,
                 from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/libSPIRV/SPIRVEnum.h:44,
                 from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/SPIRVInternal.h:43,
                 from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/VectorComputeUtil.h:43,
                 from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/VectorComputeUtil.cpp:39:
/root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/libSPIRV/SPIRVUtil.h:84:8: error:   from definition of 'template<class Ty1, class Ty2, class Identifier> void SPIRV::SPIRVMap<Ty1, Ty2, Identifier>::init()' [-fpermissive]
   void init();
        ^
/root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/VectorComputeUtil.cpp:74:69: error: specialization of 'template<class Ty1, class Ty2, class Identifier> void SPIRV::SPIRVMap<Ty1, Ty2, Identifier>::init()' in different namespace [-fpermissive]
 template <> inline void SPIRVMap<VCFloatMode, VCFloatControl>::init() {
                                                                     ^
In file included from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/libSPIRV/SPIRVOpCode.h:43:0,
                 from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/libSPIRV/SPIRVEnum.h:44,
                 from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/SPIRVInternal.h:43,
                 from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/VectorComputeUtil.h:43,
                 from /root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/VectorComputeUtil.cpp:39:
/root/parts/llvm/src/llvm/projects/SPIRV-LLVM-Translator/lib/SPIRV/libSPIRV/SPIRVUtil.h:84:8: error:   from definition of 'template<class Ty1, class Ty2, class Identifier> void SPIRV::SPIRVMap<Ty1, Ty2, Identifier>::init()' [-fpermissive]
   void init();
        ^
```

Can anyone advise what the problem is?  It's not obvious from the current LLVM Azure Pipelines setup.

--- a/ldc-config/ldc2.conf
+++ b/ldc-config/ldc2.conf
@@ -19,7 +19,6 @@ default:
     // default switches injected before all explicit command-line switches
     switches = [
         "-defaultlib=phobos2-ldc,druntime-ldc",
-        "-L--no-warn-search-mismatch",
     ];
 
     // default switches appended after all explicit command-line switches
@@ -31,11 +30,10 @@ default:
     // default directories to be searched for libraries when linking
     lib-dirs = [
         "%%ldcbinarypath%%/../lib64",
-        "%%ldcbinarypath%%/../lib32",
     ];
 
     // default rpath when linking against the shared default libs
-    rpath = "%%ldcbinarypath%%/../lib64:%%ldcbinarypath%%/../lib32";
+    rpath = "%%ldcbinarypath%%/../lib64";
 };
 
 "^wasm(32|64)-":
@@ -46,4 +44,12 @@ default:
         "-L--export-dynamic",
     ];
     lib-dirs = [];
+};
+
+"i686-.*-linux-gnu":
+{
+    lib-dirs = [
+        "%%ldcbinarypath%%/../lib32",
+    ];
+    rpath = "%%ldcbinarypath%%/../lib32";
 };

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -1,5 +1,5 @@
 name: ldc2
-version: 1.19.0
+version: 1.20.1
 summary: D compiler with LLVM backend
 description: |
     LDC is a portable compiler for the D programming Language, with
@@ -28,7 +28,7 @@ apps:
 parts:
   ldc:
     source: https://github.com/ldc-developers/ldc.git
-    source-tag: &ldc-version v1.19.0
+    source-tag: &ldc-version v1.20.1
     source-type: git
     plugin: cmake
     override-build: |

